src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs.rs

```diff
@@ -0,0 +1,534 @@
+    /// editing.
+    pub cancel_project_name_editing : frp::Source,
+    /// Sets the project name.
+    pub project_name                : frp::Source<String>,
+    /// Select the breadcrumb by its index.
+    pub select_breadcrumb           : frp::Source<usize>
+}
+
+impl FrpInputs {
+    /// Constructor.
+    pub fn new(network:&frp::Network) -> Self {
+        frp::extend! {network
+            push_breadcrumb             <- source();
+            pop_breadcrumb              <- source();
+            outside_press               <- source();
+            cancel_project_name_editing <- source();
+            project_name                <- source();
+            select_breadcrumb           <- source();
+        }
+        Self{push_breadcrumb,pop_breadcrumb,outside_press,cancel_project_name_editing,project_name,
+            select_breadcrumb}
+    }
+}
+
+/// Breadcrumbs panel frp network debug inputs. Used to test the breadcrumbs panel view without
+/// a controller.
+#[derive(Debug,Clone,CloneRef)]
+pub struct DebugFrpInputs {
+    /// Same as `FrpInputs::push_breadcrumb`, but pushes a hardcoded breadcrumb if
+    /// `Option<LocalCall>` is `None`.
+    pub push_breadcrumb   : frp::Source<Option<LocalCall>>,
+    /// Pops the breadcrumb view without sending signals to the controller.
+    pub pop_breadcrumb    : frp::Source,
+    /// Selects the breadcrumb by its index without sending signals to the controller.
+    pub select_breadcrumb : frp::Source<usize>
+}
+
+impl DebugFrpInputs {
+    /// Constructor.
+    pub fn new(network:&frp::Network) -> Self {
+        frp::extend!{ network
+            push_breadcrumb   <- source();
+            pop_breadcrumb    <- source();
+            select_breadcrumb <- source();
+        }
+        Self{push_breadcrumb,pop_breadcrumb,select_breadcrumb}
+    }
+}
+
+
+// ==================
+// === FrpOutputs ===
+// ==================
+
+/// Breadcrumbs panel frp network outputs.
+#[derive(Debug,Clone,CloneRef)]
+pub struct FrpOutputs {
+    /// Signalizes when a new breadcrumb is pushed.
+    pub breadcrumb_push   : frp::Source<Option<LocalCall>>,
+    /// Signalizes when a breadcrumb is poppoed.
+    pub breadcrumb_pop    : frp::Source,
+    /// Signalizes when project name is changed.
+    pub project_name      : frp::Any<String>,
+    /// Signalizes when a breadcrumb is selected, returning a tuple with the amount of breadcrumbs
+    /// to be popped, in case the selection happens on the left of the currently selected
+    /// breadcrumb, or else a vector of existing breadcrumbs to be pushed.
+    pub breadcrumb_select : frp::Source<(usize,Vec<Option<LocalCall>>)>
+}
+
+impl FrpOutputs {
+    /// Constructor.
+    pub fn new(network:&frp::Network) -> Self {
+        frp::extend! {network
+            breadcrumb_push   <- source();
+            breadcrumb_pop    <- source();
+            project_name      <- any_mut();
```

Comment:
use `<- (...);` instead

Reply:
@wdanilo do you mean `<- any(...)`? I don't see any usage of `<- (...)` in our codebase.